src/v2/Apps/Fair/Routes/FairExhibitors.tsx

```diff
@@ -1,46 +1,123 @@
 
+enum Mode {
+  Resting,
+  Loading,
+  Error,
+}
+
 interface FairExhibitorsProps {
   fair: FairExhibitors_fair
+  relay: RelayPaginationProp
 }
 
-const FairExhibitors: React.FC<FairExhibitorsProps> = ({ fair }) => {
+const FairExhibitors: React.FC<FairExhibitorsProps> = ({ fair, relay }) => {
+  const [mode, setMode] = useState(Mode.Resting)
+
+  const handleClick = () => {
+    if (!relay.hasMore() || relay.isLoading()) return
+
+    setMode(Mode.Loading)
+
+    relay.loadMore(15, err => {
+      if (err) {
+        setMode(Mode.Error)
+        return
+      }
+
+      setMode(Mode.Resting)
+    })
+  }
+
   return (
     <>
       {fair.exhibitors.edges.map(({ show }) => {
-        if (show.counts.artworks === 0) {
+        if (show.counts.artworks === 0 || !show.partner) {
           // Skip rendering of booths without artworks
           return null
         }
 
         return <FairExhibitorRail key={show.id} show={show} my={3} />
       })}
+
+      <Grid my={6}>
+        <Row>
+          <Col sm={6} mx="auto">
+            <Button
+              width="100%"
+              variant="secondaryGray"
+              onClick={handleClick}
+              loading={mode === Mode.Loading}
+              disabled={!relay.hasMore()}
+            >
+              Show more
+            </Button>
+          </Col>
+        </Row>
+      </Grid>
     </>
   )
 }
 
-export const FairExhibitorsFragmentContainer = createFragmentContainer(
+export const FairExhibitorsFragmentContainer = createPaginationContainer(
   FairExhibitors,
   {
     fair: graphql`
-      fragment FairExhibitors_fair on Fair {
-        id
-        exhibitors: showsConnection(first: 30, sort: FEATURED_ASC) {
+      fragment FairExhibitors_fair on Fair
+        @argumentDefinitions(
+          first: { type: "Int", defaultValue: 15 }
+          after: { type: "String" }
+        ) {
+        slug
+        exhibitors: showsConnection(
+          first: $first
+          after: $after
+          sort: FEATURED_ASC
+        ) @connection(key: "FairExhibitorsQuery_exhibitors") {
           edges {
             show: node {
               id
               counts {
                 artworks
               }
+              partner {
+                ... on Partner {
+                  id
+                }
+                ... on ExternalPartner {
+                  id
+                }
+              }
               ...FairExhibitorRail_show
             }
           }
         }
       }
     `,
+  },
+  {
+    direction: "forward",
+    getConnectionFromProps({ fair: { exhibitors } }) {
+      return exhibitors
+    },
+    getVariables({ fair: { slug: id } }, { cursor: after }, { first }) {
+      return { first, after, id }
+    },
+    query: graphql`
+      query FairExhibitorsQuery($id: String!, $first: Int!, $after: String) {
+        fair(id: $id) {
+          ...FairExhibitors_fair @arguments(first: $first, after: $after)
+        }
+      }
+    `,
```

Comment:
All of these props aren't needed I believe -- ash just posted a new blogpost with some optimizations: https://artsy.github.io/blog/2020/08/31/relay-pagination-containers-demystified/